### ISSUE TITLE
problem when uploading CSV file

### ISSUE THREAD

**Issue by chenpindong** [WeblateOrg/weblate#12034]
### Describe the issue

![企业微信截图_17205199594019](https://github.com/WeblateOrg/weblate/assets/14228594/d70175d6-768e-4ff4-9e87-b7785b382a96)

Remove template if the file contains source：

` if "source" in self.store.fieldnames and not isinstance(
            template_store, CSVFormat
        ):
            self.template_store = None`

My problem：
When translation base file is xlsx，uploading the CSV file translation, the original text was modified, but it was still found and updated.

Why is it not like this：
` if "source" in self.store.fieldnames:
            self.template_store = None`
When I make this modification, the problem is solved.

May I ask if it is possible to modify it this way？

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar filed issues in this repository.

### Steps to reproduce the behavior

1.Project Settings
![image](https://github.com/WeblateOrg/weblate/assets/14228594/d3f67fde-0d30-4fc8-b504-2a3c82e0a480)

2.Warehouse files
![企业微信截图_17205212857228](https://github.com/WeblateOrg/weblate/assets/14228594/7cec8da7-6b92-4f6f-a2b1-827a6a921f40)

3.Translation to be uploaded
![企业微信截图_17205212206149](https://github.com/WeblateOrg/weblate/assets/14228594/49b836ab-4512-4ca7-80e4-0e681197d253)

### How do you run Weblate?

Docker container

**Comment by nijel** [comment]
For monolingual files, matching is expected to happen based on key/context, source is ignored. See 4cdf14489dd66ae4f5c3d6cd3f8babbe2a72e4d2 and https://github.com/WeblateOrg/weblate/issues/1547

**Comment by chenpindong** [comment]
Why is CSV format different from other format processing methods？

` if "source" in self.store.fieldnames: self.template_store = None`
Will this modification create new problems?

**Comment by nijel** [comment]
> Why is CSV format different from other format processing methods？

Because it can have varying fields, all other formats have fixed structure.

> ` if "source" in self.store.fieldnames: self.template_store = None` Will this modification create new problems?

Do you mean removing it? Yes, it will bring back issues described in https://github.com/WeblateOrg/weblate/issues/1547

**Comment by chenpindong** [comment]
We may understand it differently, it's not that we want to delete that code.

Let me rephrase the problem:
under these circumstances,
`if "source" in self.store.fieldnames and not isinstance( template_store, CSVFormat ): self.template_store = None`
I made changes to the original text when uploading, but I still managed to update the translation. However, I hope the translation cannot be found and will not be updated.
In other formats, the translation cannot be found and will not be updated.
May I ask why the CSV format is not consistent with other formats

**Comment by nijel** [comment]
Other formats have fixed set of fields available. With CSV the fields can change, thats why there is a lot of special casing for CSV. I'm always worried changing the logic here as it can easily break other workflow and we definitely lack a test coverage here (there are too many cases to cover everything).
